feat(donation-grid): Donation form grid display option #1414

Sidsector9 opened the pull request:
Closes #1414 

## Description

1. Added `[give_donation_grid]` shortcode
2. Added following attributes to the shortcode
    1. `columns` (1|2|3|4): Sets the number of grid items per row
    2. `show_goal` (true|false): Display progress bar if goal is set
    3. `show_excerpt` (true|false): Display excerpt if it is enabled from Give Settings and shortcode.
    4. `show_featured_image` (true|false): Display featured image if it is enabled from Give Settings and shortcode.
    5. `display_type` (redirect|modal): Whether to redirect to a new page _or_ to open a modal box after clicking **Donate Now**

_[Design is kept as minimal until further discussion]_

## How Has This Been Tested?
This is WIP. Only manual testing done as of 7th Feb 2018

## Screenshots:

### Desktop
![desktop view](https://snag.gy/IJYP34.jpg)

### Tablet
![tablet view](https://snag.gy/5iQD7F.jpg)

### Video demo
https://www.useloom.com/share/edd2bdcf0cef486ea4676fbe1ca180e2

DevinWalker commented on templates/shortcode-donation-grid.php:
```diff
@@ -0,0 +1,61 @@
+		} elseif( 'modal' === $atts['display_type'] ) {
+
+			// 'Donate Now' button if the 'display_type' attribute is set to 'modal'
+			printf( '<a class="grid-donate-now grid-donate-now-modal-button" data-effect="mfp-zoom-out" href="#popup-form-%1$s">%2$s</a>', get_the_ID(), apply_filters( 'donation_grid_donate_now', esc_html__( 'Donate Now', 'grid' ) ) );
```
@Sidsector9 can you make the effect a slide down/up to match how to works with the "Modal" Display Mode? You'll see it slides in and out sort of like bootstrap modals.

Sidsector9 replied:
@DevinWalker Updated the animation

DevinWalker commented:
@Sidsector9 can we tighten up the spacing, add a bit more padding to the content, make the button full width, and basically make it look a bit prettier? Also, have you cross-theme tested this much? Thanks for the video that was helpful but I'd like to see how it looks across a few themes.

![68747470733a2f2f736e61672e67792f3569514437462e6a7067](https://user-images.githubusercontent.com/1571635/35936887-6fbd460c-0bf9-11e8-8b28-0e512d5a9994.png)

Sidsector9 commented:
@DevinWalker I changed the look, I have added a little animation so that the grid items look lively, let me know you thoughts on this - https://www.useloom.com/share/127f5e3d20924d7882dcdb75bf798baf

Sidsector9 commented:
@DevinWalker The changes are done on this one as you asked. Can you review it?
